### PR TITLE
IDE-569 Syntax highlight KEL

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-569

Signed-off-by: dehilsterlexis <david.dehilster@lexisnexis.com>

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith I would like to do a walk through the code concerning the use of the boost xml archive for a new file for sample language text. The code is commented out but can be easily tested. It should work but is not. I would prefer to use the file instead of hard coding the text. I will highlight the text in a comment.

File: wlib/PreferenceDlg.cpp [hunk 1]
```diff
@@ -1265,62 +1355,191 @@ class CPrefColorDlg : public CDialogImpl<CPrefColorDlg>,
...
+	}
+
+	void LoadSampleText()
+	{
+		std::_tstring textSample = _T("");
+		bool filled = false;
+
+		if (m_langRef)
+		{
+			//m_langRef->loadSamples();
```
Comment (dehilsterlexis) [inline wlib/PreferenceDlg.cpp]: @GordonSmith here is the boost archive load I have commented out on line 1416.

File: EclEditor/LexGENERAL.cxx [hunk 2]
```diff
@@ -0,0 +1,173 @@
+// Scintilla source code edit control
+/** @file LexKEL.cxx
```
Comment (GordonSmith) [inline EclEditor/LexGENERAL.cxx]: Comment looks wrong - I would remove it in its entirety

File: EclEditor/LexGENERAL.cxx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+static const char * const generalWordLists[] = {
+            "Headers",
+            "Thor",
+            "HThor",
+            "Roxie",
+            "Local",
+            0,
+        };
+
+LexerModule lmGENERAL(SCLEX_GENERAL, ColouriseGENERALDocSensitive, "general", FoldGENERALDoc, generalWordLists);
```
Comment (GordonSmith) [inline EclEditor/LexGENERAL.cxx]: What is the goal of the General lexer?

File: EclEditor/LexKEL.cxx [hunk 4]
```diff
@@ -0,0 +1,324 @@
+// Scintilla source code edit control
+/** @file LexKEL.cxx
+ ** Lexer for C++, C, Java, and JavaScript.
+ **/
+// Copyright 1998-2005 by Neil Hodgson <neilh@scintilla.org>
+// The License.txt file describes the conditions under which this software may be distributed.
```
Comment (GordonSmith) [inline EclEditor/LexKEL.cxx]: Remove comment block

File: en_us/resource.h [hunk 5]
```diff
@@ -537,6 +537,7 @@
 #define IDC_STATIC_WUARGUMENTS          524
 #define IDC_COMBO1                      524
 #define IDC_COMBO_MODULES               524
+#define IDC_COMBO_ELEMENT_TYPE          524
```
Comment (GordonSmith) [inline en_us/resource.h]: Should it have its own ID?

File: EclEditor/LexGENERAL.cxx (hunk 3 above)
Comment (dehilsterlexis) [inline EclEditor/LexGENERAL.cxx]: The general lexer is for non-language related color selections. I tried coding the "general" color settings without the lexer. After a 1 day trip down that road without success and everything broken beyond repair and only getting worse, I decided the easiest way to implement the non-language specific color preferences was to create a "general" lexer. It solved that problem quite nicely.

**Comment by dehilsterlexis** [comment]
@GordonSmith fixed the comments in the Lexer files and answered your questions.

**Comment by GordonSmith** [comment]
@richardkchapman This is ok to pull (into master)

**Pull request merged**